### DISCUSSION TITLE
Some translated texts doesn't appear

### DISCUSSION THREAD

**Discussion by big4billy** [ihhub/fheroes2#8168] [General]
A've found an issue during testing my language translation. Some text doesn't appear in the translated language but in English. A sample for it this:

msgid "Lighthouse"
msgid_plural "Lighthouses"
msgstr[0] "Világítótorony"
msgstr[1] "Világítótornyok"

The translation is syntactically right so I'm yust askong to look after th plural texts and their appearence in translations. Anybody who experienced this problem? in other languages too?

**Comment by oleg-derevenetz** [comment]
That's because in our `ngettext()` [implementation](https://github.com/ihhub/fheroes2/blob/141c3e1e03a0e5f758be26759e84cb251f9b9c59/src/engine/translations.cpp#L435) there are no rules regarding plurals for `LOCALE_HU`.

**Comment by big4billy** [comment]
I've found another issue in the same topic. This text is for a long time already in "hu.po":

msgid ""
"The Castle improves the town's defense and increases its income to %{count} "
"gold per day."
msgstr ""
"A Vár javítja a város védelmét és naponta %{count} arannyal növeli a "
"bevételét."

The problem shown on attached picture, the translated text doesn't appear.
![Clipboard01](https://github.com/ihhub/fheroes2/assets/143273129/062dd4a3-435f-43a3-a15d-6a6813b5037d)

**Reply by zenseii** [reply]
@big4billy. This is because the English string you wrote does not match 1:1 with the text in the game. We have probably changed the English string in the source code since the time when the `hu.po` file was made and that string translated. There are several ways to go about this. Before doing anything, I suggest making sure you have a backup of your latest `hu.po` file to make sure you don't lose any important work.

**Reply by big4billy** [reply]
For testing I changed the English text to the shown form on the picture and now it's OK. Should I change the English texts manually? I already synced our branch to the newest.

**Reply by big4billy** [reply]
Now I see, the main change is from "the hero" to "your hero".

**Reply by zenseii** [reply]
Yes, we're slowly changing it to not refer to "your hero" anymore for reasons detailed in those PRs.

**Reply by big4billy** [reply]
Got it and corrected all the issues in Hungarian translations except one. The hu.po text is below:

msgid "%{skill} increases your hero's viewable area by one square."
msgid_plural "%{skill} increases your hero's viewable area by %{count} squares."
msgstr[0] "%{skill} növeli a hős látóhatárát egy egységgel."
msgstr[1] "%{skill} növeli a hős látóhatárát %{count} egységgel."

What's wrong with it? What is the newest msgid text? They really looks like the same.

![Clipboard01](https://github.com/ihhub/fheroes2/assets/143273129/f6035be7-c42f-404a-b0ed-248057380025)

**Reply by zenseii** [reply]
@big4billy. I suspect that you are using not the latest [snapshot](https://github.com/ihhub/fheroes2/blob/master/docs/INSTALL.md#snapshots-latest-builds) or something because the string you have in the screenshot is outdated. The most current one is this one:
```
msgid "%{skill} increases the hero's viewable area by one square."
msgid_plural "%{skill} increases the hero's viewable area by %{count} squares."
```

**Reply by zenseii** [reply]
Also make sure that the string isn't fuzzied.
